test/conftest.py

```diff
@@ -171,6 +177,22 @@ def pg(tmp_path_factory, cert_dir):
     pg.cleanup()
 
 
+@pytest.fixture(scope="session")
+def replica(pg, tmp_path_factory):
+    """Starts a new Postgres replica db that is shared for tests in this process"""
+    if MACOS:
+        sudo("ifconfig lo0 alias 127.0.0.2 netmask 0xff000000")
```

Comment:
We should make sure that this test is skipped for MacOS if `USE_SUDO` is not `True`. Otherwise the tests prompt for a sudo password by default.

Reply:
5b7a91e5bdbc2d87f43e94d2ae072a80d34f9e03